kibana/fleet.go

```diff
@@ -118,83 +119,60 @@ var (
 	FALSE *bool = &bf
 )
 
+type policyResp struct {
+	Item PolicyResponse `json:"item"`
+}
+
 // CreatePolicy creates a new agent policy with the given config
-func (client *Client) CreatePolicy(request AgentPolicy) (*PolicyResponse, error) {
+func (client *Client) CreatePolicy(ctx context.Context, request AgentPolicy) (r PolicyResponse, err error) {
 	reqBody, err := json.Marshal(request)
 	if err != nil {
-		return nil, fmt.Errorf("unable to marshal create policy request into JSON: %w", err)
+		return r, fmt.Errorf("unable to marshal create policy request into JSON: %w", err)
 	}
 
-	statusCode, respBody, err := client.Request(http.MethodPost, fleetAgentPoliciesAPI, nil, nil, bytes.NewReader(reqBody))
+	resp, err := client.Connection.SendWithContext(ctx, http.MethodPost, fleetAgentPoliciesAPI, nil, nil, bytes.NewReader(reqBody))
```

Comment:
Similar to my other comment on this PR, there are a handful of methods, e.g. `UnEnrollAgent`, `GetFleetServerHost`, etc. that still use the context-less `client.Request`.  It would be good to change all of these to use `client.Connection.SendWithContext` instead, for consistency.